cactus/page.py

```diff
@@ -125,37 +120,47 @@ def build(self):
 
             self.site.plugin_manager.postBuildPage(self)
 
-    def parse_context(self, data, splitChar=':'):
+    def parse_context(self, splitChar=':'):
         """
         Values like
 
         name: koen
         age: 29
 
-        will be converted in a dict: {'name': 'koen', 'age': '29'}
+        will be converted in a dict:
+
+        {'name': 'koen', 'age': '29'}
+
+        and these lines are deleted from 'self.data'
         """
 
+        # make sure that the page context is only calculated once
+        if hasattr(self, "_page_context"):
+            return self._page_context
+
         if not self.is_html():
-            return {}, data
+            return {}
 
         values = {}
-        lines = data.splitlines()
+        lines = self._data.splitlines()
         if not lines:
-            return {}, ''
+            return {}
 
         for i, line in enumerate(lines):
-
-            if not line:
-                continue
-
-            elif splitChar in line:
-                line = line.split(splitChar)
-                values[line[0].strip()] = (splitChar.join(line[1:])).strip()
-
-            else:
+            # Context lines start at the top of the file.
+            # The text start after the first empty line
+            # or at the first line without a colon
+            if not splitChar in line:
+                if not line:
+                    i += 1
                 break
 
-        return values, '\n'.join(lines[i:])
+            key, value = line.split(splitChar, 1)
+            values[key.strip()] = value.strip()
+
+        self._data = '\n'.join(lines[i:])
```

Comment:
I think that you might as well parse the context when loading the page itself. There doesn't seem to be a use case for separating those two steps.

You could probably separate the whole page loading and parsing logic from the page object itself.